### ISSUE TITLE
[bug][ux] Output from search using explicit revision -- improve failure message

### ISSUE THREAD

**Issue by jgsogo** [conan-io/conan#8563]
When searching for a recipe-revision that doesn't exist, the error message can be improved

```
conan search zlib/1.2.11@#58dde97face18c9c280e2d6093c45d15 -r conan-center
ERROR: Not found: https://conan.bintray.com/v2/conans/zlib/1.2.11/_/_/revisions/58dde97face18c9c280e2d6093c45d15/files/conanmanifest.txt. [Remote: conan-center]
```

Searching using recipe-revision for a reference that doesn't exist

```
⇒  conan search zlib/1.2.1122@#58dde97face18c9c280e2d6093c45d15 -r conan-center
ERROR: Recipe not found: _/zlib/1.2.1122/_. [Remote: conan-center]
```

Searching a reference (not a pattern) that doesn't exit:

```
⇒  conan search zlib/1.2.1122@ -r conan-center 
ERROR: Recipe not found: _/zlib/1.2.1122/_. [Remote: conan-center]
```

**Comment by ghost** [comment]
Hi, I'm interested in taking this on

**Comment by uilianries** [comment]
@Alzamer Thank you for your intention, please, go ahead and comment this issue in your PR when opening.

**Comment by MariaMozgunova** [comment]
Is this issue already fixed?

**Comment by nreichen** [comment]
Hello world!👋🌎
I would like to take on this issue if that is okay. Please expect a pull request coming from me shortly.

**Comment by memsharded** [comment]
Hi @nreichen 

Thanks very much for your offer.
Please note that Conan 2.0 is out. This PR should be targeted to ``release/2.0`` branch to be part of the next 2.0.X patch (this would be part of the ``conan list`` command in 2.0)

**Comment by nreichen** [comment]
Hi @memsharded 

Thanks so much for getting back to me.

You said that "This PR should be targeted to release/2.0 branch to be part of the next 2.0.X patch."

However, in the [Contributing to Conan](https://github.com/conan-io/conan/blob/dc17e1c113ef2773a23a25e317bd3d218aa02a9a/.github/CONTRIBUTING.md) docs, step 5 states "5. Open a pull request, and select develop as the base branch. Never open a pull request to release/xxx branches."

I'm a bit confused on which branch I should open a PR to.

Thanks in advance.

**Comment by memsharded** [comment]
Yes, that doc is a bit outdated, for Conan 2.0:

- ``develop2`` is the branch for the next 2.X release (at the moment only for next 2.0.X)
- ``release/2.0`` is the "live" branch of the latest 2.0.X docs, at the moment 2.0.2

**Comment by nreichen** [comment]
@memsharded thanks for the clarification.

So just to confirm, I should be making a PR to `release/2.0`?

**Comment by memsharded** [comment]
Yes, please ``release/2.0`` branch

**Comment by kidkoder432** [comment]
This issue appears to have been solved by #14309. Should it be closed?

**Comment by memsharded** [comment]
Yes, thanks for the follow up @kidkoder432 !
